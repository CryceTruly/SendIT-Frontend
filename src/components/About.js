import React, {
  Component
} from 'react'

export default class About extends Component {
  render() {
    return (
      <div>
      <h1> This is about </h1> </div>
    )
  }
}