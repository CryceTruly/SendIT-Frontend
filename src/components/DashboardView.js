import React from 'react'

function DashboardView() {
  return (
    <div>
        <h3>Stats</h3>

    </div>
  )
}

export default DashboardView
