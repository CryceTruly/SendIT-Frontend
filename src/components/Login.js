import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import LoginView from "../views/Login";
import { clearErrors } from "../actions/errorActions";
import { login } from "../actions/authActions";

export class Login extends Component {
  state = {
    email: "",
    password: "",
    msg: null,
  };

  componentDidUpdate(prevProps) {
    const { error } = this.props;
    if (error !== prevProps.error) {
      // Check for auth error
      /* istanbul ignore next */
      if (error.id === "LOGIN_FAIL") {
        this.setState({ msg: error.msg });
      } else {
        this.setState({ msg: null });
      }
    }
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = (e) => {
    e.preventDefault();

    const { email, password } = this.state;

    // Create user data
    const newUser = {
      email,
      password,
    };

    // Attempt to login
    this.props.login(newUser, this.props);
  };

  render() {
    return (
      <LoginView
        msg={this.state.msg}
        onChange={this.onChange}
        onSubmit={this.onSubmit}
        errorMessage={this.props.error.msg}
        errorId={this.props.error.id}
        processing={this.props.error.processing}
      />
    );
  }
}
export const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
  error: state.errors,
  auth: state.auth,
});

export default connect(
  mapStateToProps,
  { login, clearErrors },
)(Login);
