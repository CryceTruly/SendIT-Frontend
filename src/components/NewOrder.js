import React, { Component } from 'react'
import {
    Button,
    Card,
    CardBody,
    CardTitle,
    Form,
    FormGroup,
    Label,
    Input,
    Alert,
    Row,
    Col
  } from 'reactstrap';
export default class NewOrder extends Component {
    state = {
        fullname: '',
        email: '',
        password: '',
        phone_number:'',
        username:'',
        msg: null
      };

  render() {
    return (
      <div>

         <Card>
        <CardBody>
          <CardTitle></CardTitle>

          <CardTitle><strong>Add a new Parcel Request.</strong></CardTitle>


            <Form onSubmit={this.onSubmit}>


{this.state.msg?<Alert color="danger">{this.state.msg}</Alert>:null}

            <Row>
              <div className="col-md-6">

              <FormGroup>
                <Label for='fullname'>Pickup Location</Label>
                <Input
                  type='text'
                  name='fullname'
                  id='fullname'
                  placeholder='Full Name' required
                  className='mb-3'
                  onChange={this.onChange}
                />
                </FormGroup>


<FormGroup>
                <Label for='email'>Destination address</Label>
                <Input
                  type='email'
                  name='email'
                  id='email' required
                  placeholder='Email'
                  className='mb-3'
                  onChange={this.onChange}/>
                 </FormGroup>
                <FormGroup>
<Label for='phone_number'>Recipient Phone Number</Label>
                <Input
                  type='text'
                  name='phone_number'
                  id='phone_number' required
                  placeholder='phone_number'
                  className='mb-3'
                  onChange={this.onChange}
                />
                </FormGroup>

              </div>
               <div className="col-md-6">
<FormGroup>
  <Label for='username'>Recipient Fullname</Label>
                <Input
                  type='text'
                  name='username'
                  id='username' required
                  placeholder=''
                  className='mb-3'
                  onChange={this.onChange}
                />
</FormGroup>
<FormGroup>
   <Label for='password'>Parcel Description</Label>
                <Input
                  type='password'
                  name='password' required
                  id='password'
                  placeholder=''
                  className='mb-3'
                  onChange={this.onChange}
                />



</FormGroup>

<FormGroup>

                  <Label for='password_comfirm'>Weight </Label>
                <Input
                  type='password_comfirm'
                  name='password'
                  id='password_comfirm' required
                  placeholder=''
                  className='mb-3'
                  onChange={this.onChange}
                />
</FormGroup>
 <Button color='dark' style={{ marginTop: '2rem' }} block>
                  Submit
                </Button>


               </div>


            </Row>




            </Form>
            </CardBody>
            </Card>



      </div>
    )
  }
}
