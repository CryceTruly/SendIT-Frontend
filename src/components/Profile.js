import React, { Component } from 'react'
import {Button,Card,CardImg,CardBody,CardSubtitle,CardText,CardTitle} from 'reactstrap';
import {connect} from 'react-redux';
class Profile extends Component {

    getStyle=()=>{
        return{
            color: 'blue',
            border:'2px solid black'
        }
    }
  render() {

    const { isAuthenticated,user } = this.props.auth;

    if(this.props.auth.isLoading){
       return <h3>Loading User...</h3>
    }else{
        const {user} = this.props.auth;




return(
        <center>
        <div className="card mx-auto">
          <div className="card-body pt-5">
            <img src="https://avatars1.githubusercontent.com/u/20795487?s=400&v=4" height="200" className="rounded-circle" alt="profile-image"/>
             <p className="card-text">   FullName:  <strong>{user ? `${user.fullname}` : ''}</strong></p>
             <p className="card-text">  Username:   <strong>{user ? `${user.username}` : ''}</strong></p>
            <p className="card-text">Email:<strong>{user ? `${user.email}` : ''}</strong></p>
            <p className="card-text">Joined:<strong>{user ? `${user.joined}` : ''}</strong></p>
            <Button className="btn-success">Edit  Profile</Button>

</div>

</div>

<br>
</br> <div>
      <Card>
       <CardBody>
          <CardTitle>Statistics</CardTitle>
        </CardBody>

        <h6>All orders <span className="badge badge-info">5</span></h6><h6>Delivered <span  className="badge badge-info">5</span></h6><h6 >Undelivered <span className="badge badge-info">5</span></h6>
      </Card>
    </div>


</center>
  )
}
}
}
const mapStateToProps=(state)=>({

  auth: state.auth,
    user:state.auth.user
});
export default connect(mapStateToProps,{})(Profile)