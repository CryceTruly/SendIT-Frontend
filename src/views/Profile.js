import React from "react";
import PropTypes from "prop-types";
import {
  Button, Card, CardImg, CardBody, CardSubtitle, CardText, CardTitle,
} from "reactstrap";

const ProfileView = (props) => {
  const { user, parcels } = props;
  return (
    <center>
      <div className="card mx-auto">
        <div className="card-body pt-5">
          <img src={user ? user.imageUrl : ""} alt="" height="150" className="rounded-circle my-4" />
          <p className="card-text">
            {" "}
FullName:
            <strong>{user ? user.fullname : ""}</strong>
          </p>
          <p className="card-text">
            {" "}
Username:
            <strong>{user ? user.username : ""}</strong>
          </p>
          <p className="card-text">
Email:
            <strong>{user ? user.email : ""}</strong>
          </p>
          <p className="card-text">
Joined:
            <strong>{user ? user.joined : ""}</strong>
          </p>
          <Button className="btn-success">Edit  Profile</Button>
        </div>
      </div>
      <br />
      {" "}
      <div>
        <Card>
          <CardBody>
            <CardTitle>Statistics</CardTitle>
          </CardBody>
          <h6>
All orders
            <span className="badge badge-info">
              {parcels.length
              }
            </span>
          </h6>
          <h6>
Delivered
            <span className="badge badge-info">
              {parcels.filter(parcel => parcel.status === "delivered").length
              }
            </span>
          </h6>
          <h6>
Pending
            <span className="badge badge-info">
              {parcels.filter(parcel => parcel.status === "order_placed").length
              }
            </span>
          </h6>
        </Card>
      </div>
    </center>
  );
};

export default ProfileView;
