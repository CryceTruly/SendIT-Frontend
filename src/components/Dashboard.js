import React, { Component } from "react";
import DashboardView from "./DashboardView";

export default class Dashboard extends Component {
  render() {
    return (

      <h3>Welcome to SendIT Adminsration</h3>

    );
  }
}
