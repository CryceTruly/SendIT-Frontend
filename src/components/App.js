import React, {
    Component
} from "react";
import {
    Provider
} from "react-redux";
import {
    Container
} from "reactstrap";
import {
    BrowserRouter as Router, Switch, Link, Route
} from "react-router-dom";
import store from "../Store";
import Parcels from "./Parcels";
import Navbar from "./shared/Navbar";
import {
    loadUser
} from "../actions/authActions";
import Login from "./Login";
import RegisterComponent from "./Register";
import PrivateRoute from "./common/PrivateRoute";

export class App extends Component {

    componentDidMount(){
        store.dispatch(loadUser());
    }

    render() {
        return (
            <Router>
                <Provider store={store}>

                    <Navbar />
                    <Container>
                        <p />

                        <PrivateRoute exact path="/" component={Parcels} />
                        <Route exact path="/login" component={Login} />
                        <Route exact path="/parcels" component={Parcels} />
                        <Route exact path="/register" component={RegisterComponent} />
                    </Container>
                </Provider>
            </Router>





        );
    }
}
export default App;
