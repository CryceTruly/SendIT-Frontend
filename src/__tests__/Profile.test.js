import React from 'react';
import {shallow} from 'enzyme';
import Profile from '../components/Profile'
import {mapStateToProps} from '../components/Profile'
describe(`Profile component`,()=>{
    it('should display user details',async ()=>{
        const wrapper = shallow(<Profile />);
        expect(wrapper).toMatchSnapshot()


    })


    it(`should  map parcel state to props`,()=>{
        const parcels=[
            {
                destination_address: "Uganda",
                parcel_id: 7,
                pickup_address: "Ntungamo - Katunguru Rd, Uganda",
                placed: "Tue, 26 Mar 2019 11:12:18 GMT",
                recipient_email: "Ntungamo - Katunguru Rd, Uganda",
                recipient_phone_number: "0766765423",
                sender_email: "aacryce@gmail.com",
                status: "order_placed",
                user_id: 11
            }
        ]
    const appState={"parcels":parcels}
    const componentState=mapStateToProps({"parcels":appState});
    expect(componentState).toEqual(appState)


    });
})