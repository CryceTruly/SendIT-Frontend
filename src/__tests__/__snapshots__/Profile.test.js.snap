// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should display user details 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    auth={
      Object {
        "isLoading": false,
        "user": Object {
          "username": "hello",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "parcels": undefined,
      "user": Object {
        "username": "hello",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "parcels": undefined,
        "user": Object {
          "username": "hello",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Profile component should show loading of a user 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    auth={
      Object {
        "isLoading": true,
        "user": Object {
          "username": "hello",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading User...",
    },
    "ref": null,
    "rendered": "Loading User...",
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading User...",
      },
      "ref": null,
      "rendered": "Loading User...",
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
